service/collector/error.go

```diff
@@ -1,12 +1,56 @@
 package collector
 
 import (
+	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/giantswarm/microerror"
 )
 
+const (
+	// trustedAdvisorSubscriptionRequiredExceptionCode is the error code returned
+	// if Trusted Advisor is not supported (support plan is not Business or Enterprise).
+	trustedAdvisorSubscriptionRequiredExceptionCode = "SubscriptionRequiredException"
+)
+
 var invalidConfigError = microerror.New("invalid config")
 
 // IsInvalidConfig asserts invalidConfigError.
 func IsInvalidConfig(err error) bool {
 	return microerror.Cause(err) == invalidConfigError
 }
+
+var invalidResourceError = microerror.New("invalid resource")
+
+// IsInvalidResource asserts invalidResourceError.
+func IsInvalidResource(err error) bool {
+	return microerror.Cause(err) == invalidResourceError
+}
+
+var nilLimitError = microerror.New("nil limit")
+
+// IsNilLimit asserts nilLimitError.
+func IsNilLimit(err error) bool {
+	return microerror.Cause(err) == nilLimitError
+}
+
+var nilUsageError = microerror.New("nil usage")
+
+// IsNilUsage asserts nilUsageError.
+func IsNilUsage(err error) bool {
+	return microerror.Cause(err) == nilUsageError
+}
+
+// IsUnsupportedPlan asserts that an error is due to Trusted Advisor
+// not being available with the current support plan.
+func IsUnsupportedPlan(err error) bool {
```

Comment:
We always have pairs or error and matchers. The matchers are always handling our custom errors. The matcher often get extended to match based on other conditions. Our custom error and its handling within its matcher is still missing. The `fmt` reference describes all of this. The examples I shared also show all of this.

Reply:
Does that mean creating unused variable? Doesn't sound good IMO.

Reply:
> in theory the error should always be used.

When it doesn't exists it won't be matched. That's 100%. I don't follow why creating additional noise.

Reply:
I made my point and unsubscribe from this conversation now.